Incorporate review feedback from OpenOCD team

timsifive opened the pull request:
Remove unused file handle, and rename the only remaining one to make
more sense.
Close file descriptor if initialization fails.
Review: http://openocd.zylin.com/#/c/4312/
Change-Id: I383567aaadb1aa59d86f814eba8bc65f24e91928

timsifive committed:
Incorporate review feedback from OpenOCD team

Remove unused file handle, and rename the only remaining one to make
more sense.
Close file descriptor if initialization fails.

Change-Id: I383567aaadb1aa59d86f814eba8bc65f24e91928

palmer-dabbelt commented on src/jtag/drivers/remote_bitbang.c:
```diff
@@ -289,17 +288,10 @@ static int remote_bitbang_init(void)
-
-	remote_bitbang_out = fdopen(remote_bitbang_fd, "w");
-	if (remote_bitbang_out == NULL) {
+	remote_bitbang_file = fdopen(remote_bitbang_fd, "w");
```
I thought you needed "w+" to open for both read and write?

timsifive replied:
You're right. It's been this way forever so clearly it works, but "w+" is more correct. I fixed it.

timsifive committed:
fdopen() for writing and reading.

Since that's what we're doing. Doesn't seem like this affected anything,
though.

Change-Id: I0fa915fe2a311d4b3e6e72a9a9a19f52018258c3

timsifive committed:
Format comments to be doxygen style.

Change-Id: I7a757b92926b9cd65846551893f78ffb5e462370

timsifive committed:
Propagate errors from remote_bitbang; don't exit()

Change requested by Paul Fertser at
http://openocd.zylin.com/#/c/4312/4/src/jtag/drivers/remote_bitbang.c

Change-Id: I44d97bd1198821d2e8afdc7a237ed3c3825cd319
Signed-off-by: Tim Newsome <tim@sifive.com>

timsifive committed:
Update parport to return errors.

Change-Id: I6c6bf75809cd3222c7680e10ac6ee6073050ed07

timsifive committed:
Fix 2 more bitbang drivers.

Change-Id: Ib7257d1d113871a9f57ba3b899cb029bb595035a

timsifive commented:
This just merged upstream, so no need to separately merge it here as well. I'll take care of it when we pull from upstream again. (Hopefully this week.)
